wakuv2/gowaku.go

```diff
@@ -549,22 +549,44 @@ func (w *Waku) discoverAndConnectPeers() {
 					w.logger.Error("could not obtain dns discovery peers for ClusterConfig.WakuNodes", zap.Error(err), zap.String("dnsDiscURL", addrString))
 				}
 			}()
-		} else {
+			continue
+		}
+
+		if addr, err := multiaddr.NewMultiaddr(addrString); err == nil {
 			// It is a normal multiaddress
-			addr, err := multiaddr.NewMultiaddr(addrString)
+			peerInfo, err := peer.AddrInfoFromP2pAddr(addr)
 			if err != nil {
-				w.logger.Warn("invalid peer multiaddress", zap.String("ma", addrString), zap.Error(err))
+				w.logger.Warn("invalid peer multiaddress", zap.Stringer("addr", addr), zap.Error(err))
 				continue
 			}
 
-			peerInfo, err := peer.AddrInfoFromP2pAddr(addr)
+			go w.connect(*peerInfo, nil, wps.Static)
+			continue
+		}
+
+		if strings.HasPrefix(addrString, "enr:") {
```

Comment:
It seems more clear if move this condition before `multiaddr.NewMultiaddr(addrString)`.